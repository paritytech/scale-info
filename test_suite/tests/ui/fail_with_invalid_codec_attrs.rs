use scale_info::TypeInfo;
use scale::Encode;

#[derive(TypeInfo, Encode)]
struct NoMultipleAttrs {
    a: u8,
    #[codec(skip, compact)]
    b: u16,
}

#[derive(Encode, TypeInfo)]
enum NoIndexOnVariantFields {
    Thing(#[codec(index = 3)] u32),
}

#[derive(Encode, TypeInfo)]
enum IndexMustBeNumber {
    #[codec(index = a)]
    Thing(u32),
}

#[derive(Encode, TypeInfo)]
enum EncodeAsAttrMustBeLiteral {
    #[codec(encode_as = u8, compact)]
    Thong(bool),
}

fn main() {}
