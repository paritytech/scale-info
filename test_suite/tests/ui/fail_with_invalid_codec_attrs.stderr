error: Invalid attribute on field, only `#[codec(skip)]`, `#[codec(compact)]` and `#[codec(encoded_as = "$EncodeAs")]` are accepted.
 --> $DIR/fail_with_invalid_codec_attrs.rs:7:7
  |
7 |     #[codec(skip, compact)]
  |       ^^^^^^^^^^^^^^^^^^^^

error: Invalid attribute on field, only `#[codec(skip)]`, `#[codec(compact)]` and `#[codec(encoded_as = "$EncodeAs")]` are accepted.
  --> $DIR/fail_with_invalid_codec_attrs.rs:13:19
   |
13 |     Thing(#[codec(index = 3)] u32),
   |                   ^^^^^^^^^

error: expected literal
  --> $DIR/fail_with_invalid_codec_attrs.rs:18:21
   |
18 |     #[codec(index = a)]
   |                     ^

error: expected literal
  --> $DIR/fail_with_invalid_codec_attrs.rs:24:25
   |
24 |     #[codec(encode_as = u8, compact)]
   |                         ^^
